crates/tests/tedge_test_utils/src/fs.rs

```diff
@@ -60,6 +60,32 @@ impl TempTedgeDir {
         TempTedgeFile { file_path: path }
     }
 
+    #[cfg(unix)]
+    pub fn link_dir(&self, original: &TempTedgeDir, link_name: &str) -> TempTedgeDir {
+        let root = Utf8Path::from_path(self.temp_dir.path()).unwrap();
+        let path = root.join(&self.current_file_path).join(link_name);
+
+        std::os::unix::fs::symlink(original.path(), &path).unwrap();
+
+        TempTedgeDir {
+            temp_dir: self.temp_dir.clone(),
+            current_file_path: path,
+        }
+    }
+
+    #[cfg(windows)]
+    pub fn link_dir(&self, original: &TempTedgeDir, link_name: &str) -> TempTedgeDir {
+        let root = Utf8Path::from_path(self.temp_dir.path()).unwrap();
+        let path = root.join(&self.current_file_path).join(link_name);
+
+        std::os::windows::fs::symlink_dir(original.path(), &path).unwrap();
```

Comment:
This test in windows might be problematic as on IT managed windows laptops, generally the default seems to be to disallow creation of symlinks (without admin rights)...Is there someway we can mark this test as optional, so it does not run by default when running on windows?

Reply:
We've decided to include the test by default rather than hiding it behind an opt-in feature (as this would lead to the test being too easily forgotten).

Running the tests on public Github runner should not be affected by the admin rights issue.